pkg/cloud/rgraph/exec/executor_parallel.go

```diff
@@ -104,6 +100,42 @@ func (ex *parallelExecutor) Run(ctx context.Context) (*Result, error) {
 
 }
 
+func (ex *parallelExecutor) deepCopyResult() *Result {
```

Comment:
Put the copy method on Result itself (without lock)

```
func (*Result) copy() *Result {
 ret := Result{
		Completed: make([]Action, len(ex.result.Completed)),
		Pending:   make([]Action, len(ex.result.Pending)),
		Errors:    make([]ActionWithErr, len(ex.result.Errors)),
	}
	copy(resultCopy.Completed, ex.result.Completed)
	copy(resultCopy.Errors, ex.result.Errors)
	copy(resultCopy.Pending, ex.result.Pending)
  return ret
}
```

Reply:
done